sklearn/pipeline.py

```diff
@@ -221,22 +227,58 @@ def _validate_steps(self):
                             "'%s' (type %s) doesn't"
                             % (estimator, type(estimator)))
 
+        return deepcopy(self._steps)
```

Comment:
I am completely fine with your change which is more explicit.
I was thinking that `deepcopy` would trigger the same behaviour than what you are proposing. Am I missing something regarding the behaviour of `deepcopy`?